File: Jewels-and-Stones/jewels-and-stones.cpp
#include <string>
class Solution {
public:
    int numJewelsInStones(std::string jewels, std::string stones) {
        int ans = 0;
        for(char i : jewels){
            for(char j : stones){
                if(i==j){
                    ans++;
                }
            }
        }
        return ans;
    }
};

// simpler to code, using the c++ count function.
class Solution2{
    public: int numJewelsInStones(std::string jewels, std::string stones){
        int ans = 0;
        for(char i : jewels){
            ans += count(stones.begin(),stones.end(),i);
        }
        return ans;
    }
};

File: Jewels-and-Stones/jewels-and-stones.py
class Solution:
    def numJewelsInStones(self, jewels: str, stones: str) -> int:
        c = 0
        for j in jewels:
            for s in stones:
                if j == s:
                    c+=1
        return c

class Solution2: # simpler to code. using count function
    def numJewelsInStones(self, jewels: str, stones: str) -> int:
        ans = 0
        for x in jewels:
            ans += stones.count(x)
        return ans
        
            